master/internal/kubernetes/pod.go

```diff
@@ -496,30 +502,67 @@ func (p *pod) configureVolumes(
 }
 
 func (p *pod) configurePodSpec(
+	ctx *actor.Context,
 	volumes []k8sV1.Volume,
 	initContainers []k8sV1.Container,
 	containers []k8sV1.Container,
+	podSpec *k8sV1.Pod,
 ) *k8sV1.Pod {
-	return &k8sV1.Pod{
-		ObjectMeta: metaV1.ObjectMeta{
-			Name:      p.podName,
-			Namespace: p.namespace,
-			Labels:    map[string]string{determinedLabel: p.taskSpec.TaskID},
-		},
-		Spec: k8sV1.PodSpec{
-			Volumes:        volumes,
-			HostNetwork:    p.taskSpec.TaskContainerDefaults.NetworkMode.IsHost(),
-			InitContainers: initContainers,
-			Containers:     containers,
-			RestartPolicy:  k8sV1.RestartPolicyNever,
-		},
+	if podSpec == nil {
+		podSpec = &k8sV1.Pod{}
+	} else {
+		podSpec = podSpec.DeepCopy()
+	}
+
+	podSpec.ObjectMeta.Name = p.podName
+	podSpec.ObjectMeta.Namespace = p.namespace
+	if podSpec.ObjectMeta.Labels == nil {
+		podSpec.ObjectMeta.Labels = make(map[string]string)
 	}
+	podSpec.ObjectMeta.Labels[determinedLabel] = p.taskSpec.TaskID
+
+	if len(podSpec.Spec.Containers) > 0 {
+		for k, v := range podSpec.Spec.Containers[0].Resources.Limits {
+			if _, present := containers[0].Resources.Limits[k]; !present {
+				containers[0].Resources.Limits[k] = v
+			}
+		}
+
+		for k, v := range podSpec.Spec.Containers[0].Resources.Requests {
+			if _, present := containers[0].Resources.Requests[k]; !present {
+				containers[0].Resources.Requests[k] = v
+			}
+		}
+
+		containers[0].VolumeMounts = append(
+			containers[0].VolumeMounts, podSpec.Spec.Containers[0].VolumeMounts...)
+
+		containers[0].VolumeDevices = append(
+			containers[0].VolumeDevices, podSpec.Spec.Containers[0].VolumeDevices...)
+	}
+
+	podSpec.Spec.Volumes = append(podSpec.Spec.Volumes, volumes...)
+	podSpec.Spec.HostNetwork = p.taskSpec.TaskContainerDefaults.NetworkMode.IsHost()
+	podSpec.Spec.InitContainers = initContainers
+	podSpec.Spec.Containers = containers
+	podSpec.Spec.RestartPolicy = k8sV1.RestartPolicyNever
+
+	ctx.Log().Debugf("launching pod spec %v", podSpec.Spec)
```

Comment:
blocking: can you move this to "launchPod"

Reply:
ah sorry kept forgetting to do this